clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -6491,14 +6488,8 @@ llvm::DILocation *CodeGenFunction::SanitizerAnnotateDebugInfo(
   else
     Label = SanitizerHandlerToCheckLabel(Handler);
 
-  for (auto Ord : Ordinals) {
-    // TODO: deprecate ClArrayBoundsPseudoFn
-    if (((ClArrayBoundsPseudoFn && Ord == SanitizerKind::SO_ArrayBounds) ||
-         CGM.getCodeGenOpts().SanitizeAnnotateDebugInfo.has(Ord)) &&
-        CheckDebugLoc) {
-      return DI->CreateSyntheticInlineAt(CheckDebugLoc, Label);
-    }
-  }
+  if (any_of(Ordinals, [&](auto Ord) { return AnnotateDebugInfo.has(Ord); }))
```

Comment:
Maybe `!any_of` could be combined with the `AnnotateDebugInfo.empty()` check earlier? That would avoid the need for `return CheckDebugLoc;` in line 6494.

Reply:
Doesn't that kind of obfuscate the code? Right now, we have preconditions / early exit first, then the actual logic "if any Ord matches"